specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -4996,6 +5703,320 @@
         }
       }
     },
+    //////////////////////////////////////////////////////
+    // Inference Endpoint
+    //////////////////////////////////////////////////////
+    "InferenceEndpointResource": {
+      "type": "object",
+      "description": "Machine Learning Inference Endpoint object wrapped into ARM resource envelope.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "description": "Inference Endpoint properties",
+              "properties": {
+                "description": {
+                  "description": "The Endpoint description.",
+                  "type": "string"
+                },
+                "properties": {
+                  "description": "The Endpoint property dictionary. Properties are immutable.",
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                },
+                "inferenceEndpointProvisioningState": {
+                  "type": "string",
+                  "enum": [
+                    "Unknown",
+                    "Creating",
+                    "Deleting",
+                    "Succeeded",
+                    "Failed",
+                    "Canceled"
+                  ],
+                  "x-ms-enum": {
+                    "name": "InferenceEndpointProvisioningState",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "The current provisioning state of the Inference Deployment resource."
+                },
+                "error": {
```

Comment:
Is this the errorMessage when state is Failed? Should we call it statusMessage or failureMessage? is there a comparable we can reference to decide?

Reply:
this was a property left over from copying service. Compute has ProvisioningError that holds the error and AmlCompute keeps an array of error encountered for compute nodes during node setup. We can keep this property for failed endpoints.